libs/partners/elasticsearch/langchain_elasticsearch/vectorstores.py

```diff
@@ -70,6 +70,7 @@ def index(
         self,
         dims_length: Union[int, None],
         vector_query_field: str,
+        text_field: str,
```

Comment:
Suggestion, this could be `str | str[]`, and when a list is provided a `multi_match` query is used.

Reply:
this is fine as a single field as developers are not able to search on multiple fields within a VectorStore.

Reply:
As @joemcelroy mentioned, this is because of the limitations of the `VectorStore` and `BaseRetrievalStrategy`. I'm also hoping for the `ElasticsearchStore` to support multi-match searches in the future. But, let's keep that outside this PR for now and think about it as a future work.